tests/components/buienradar/test_sensor.py

```diff
@@ -44,5 +119,122 @@ async def test_smoke_test_setup_component(
     await hass.async_block_till_done()
 
     for cond in CONDITIONS:
-        state = hass.states.get(f"sensor.buienradar_5_40021651_528850{cond}")
+        state = hass.states.get(f"sensor.buienradar_51_5288505_400216{cond}")
         assert state.state == "unknown"
+
+
+async def test_smoke_test_data(
+    aioclient_mock: AiohttpClientMocker,
+    hass: HomeAssistant,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Smoke test for successfully set-up with default configand data."""
+    brJson = load_json_object_fixture("buienradar.json", DOMAIN)
+    raindata = load_fixture("raindata.txt", DOMAIN)
+    aioclient_mock.get(
+        "https://data.buienradar.nl/2.0/feed/json", status=HTTPStatus.OK, json=brJson
+    )
+    aioclient_mock.get(
+        f"https://gpsgadget.buienradar.nl/data/raintext?lat={TEST_CFG_DATA[CONF_LATITUDE]:.2f}&lon={TEST_CFG_DATA[CONF_LONGITUDE]:.1f}",
+        status=HTTPStatus.OK,
+        text=raindata,
+    )
+    mock_entry = MockConfigEntry(domain=DOMAIN, unique_id="TEST_ID", data=TEST_CFG_DATA)
+
+    mock_entry.add_to_hass(hass)
+
+    for cond in CONDITIONS:
+        entity_registry.async_get_or_create(
+            domain="sensor",
+            platform="buienradar",
+            unique_id=f"{TEST_CFG_DATA[CONF_LATITUDE]:2.6f}{TEST_CFG_DATA[CONF_LONGITUDE]:2.6f}{cond}",
+            config_entry=mock_entry,
+            original_name=f"Buienradar {cond}",
+        )
+    await hass.async_block_till_done()
```

Comment:
The integration creates the entities so is there any reason to do this prior to setup?

Reply:
Addec ocde comments as to why this is done (due to the platform not registering these sensors by default)

Reply:
You should use the fixture `entity_registry_enabled_by_default` which will enable all the entities before the tests instead of creating them manually in each test.